### PR TITLE
Adding boolean argument to ML Commons API

### LINKED ISSUE #135: [FEATURE]Adding boolean async argument to ML Commons API
**Is your feature request related to a problem?**
I noticed that some API calls take a bit of time, like upload_pretrained_model, load_model, unload_model, and _send_model_info. get_task_info already has such an argument called wait_until_task_done. I think the user should have the option of making an asynchronous API call.

**Do you have any additional context?**
@dhrubo-os If this makes sense, I'd like to work on this. The implementation should be simple.

### PR DESCRIPTION
**Pull request by AlibiZhenis**
### Description
Closes #135 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch_py_ml/ml_commons/ml_commons_client.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def upload_pretrained_model(
         model_version: str,
         model_format: str,
         load_model: bool = True,
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_commons/ml_commons_client.py:71]: Cool!! Can we also add this functionality in the `upload_model` also?

File: opensearch_py_ml/ml_commons/ml_common_utils.py [hunk 2]
```diff
@@ -9,3 +9,4 @@
 MODEL_UPLOAD_CHUNK_SIZE = 10_000_000
 MODEL_MAX_SIZE = 4_000_000_000
 BUF_SIZE = 65536  # lets read stuff in 64kb chunks!
+TIMEOUT = 120  # timeout for synchronous method calls in seconds
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_commons/ml_common_utils.py:12]: thanks for putting this here. Can you please make sure all the `120` number is replaced by this variable? Thanks.

File: opensearch_py_ml/ml_commons/ml_commons_client.py [hunk 3]
```diff
@@ -29,7 +29,9 @@ def upload_model(
         self,
         model_path: str,
         model_config_path: str,
+        load_model: bool,
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_commons/ml_commons_client.py:32]: `load_model` default value `false`.

File: opensearch_py_ml/ml_commons/ml_commons_client.py [hunk 4]
```diff
@@ -56,19 +58,30 @@ def upload_model(
...
+
+        return model_id
+
     def upload_pretrained_model(
         self,
         model_name: str,
         model_version: str,
         model_format: str,
-        load_model: bool = True,
+        load_model: bool,
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_commons/ml_commons_client.py:83]: same here.

**Review by dhrubo-os — APPROVED** [review]

File: opensearch_py_ml/ml_commons/ml_commons_client.py [hunk 5]
```diff
@@ -56,19 +58,30 @@ def upload_model(
         :type model_config_path: string
         :param isVerbose: if isVerbose is true method will print more messages. default False
         :type isVerbose: boolean
+        :param load_model: Whether to load the model in memory using uploaded model chunks
```
Comment (ylwu-amzn) [inline opensearch_py_ml/ml_commons/ml_commons_client.py:61]: Not related to this PR, just remind something we should do.
From 2.7 we are going to rename the APIs https://github.com/opensearch-project/ml-commons/issues/823. We should also change in py-ml

**Review by ylwu-amzn — APPROVED** [review]
Thanks for the contribution!

File: opensearch_py_ml/ml_commons/ml_commons_client.py (hunk 5 above)
Comment (dhrubo-os) [reply opensearch_py_ml/ml_commons/ml_commons_client.py:61]: Yeah, we have issue open for this: https://github.com/opensearch-project/opensearch-py-ml/issues/122

**Pull request merged**